liberum_gui/src/system_observer.rs

```diff
@@ -0,0 +1,152 @@
+use liberum_core::{node_config::NodeConfig, types::NodeInfo};
+use std::{
+    collections::{HashMap, HashSet},
+    path::Path,
+    sync::Arc,
+    time::Duration,
+};
+
+use anyhow::{anyhow, Result};
+use liberum_core::{DaemonRequest, DaemonResponse, DaemonResult};
+use tokio::sync::mpsc::{Receiver, Sender};
+
+use std::sync::Mutex;
+use tracing::{debug, error};
+
+#[derive(Default, Clone)]
+pub struct SystemState {
+    pub node_infos: Vec<NodeInfo>,
+    pub node_configs: HashMap<String, NodeConfig>,
+}
+
+pub struct SystemObserver {
+    rt: tokio::runtime::Runtime,
+    pub system_state: Arc<Mutex<Option<SystemState>>>,
+    to_daemon_sender: Option<Sender<DaemonRequest>>,
+    from_daemon_receiver: Option<Receiver<DaemonResult>>,
+    observed_node_configs: Arc<Mutex<HashSet<String>>>,
+}
+
+impl SystemObserver {
+    pub fn new() -> Result<Self> {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(1)
+            .enable_all()
+            .build()?;
+        let path = Path::new("/tmp/liberum-core/");
+        let contact =
+            rt.block_on(async { liberum_core::connect(path.join("liberum-core-socket")).await });
+        let (to_daemon_sender, from_daemon_receiver) = match contact {
+            Ok(c) => c,
+            Err(e) => {
+                error!(
+                    err = e.to_string(),
+                    "Failed to connect to the core. Make sure the core is running!"
+                );
+                Err(anyhow!(e))?
+            }
+        };
+
+        Ok(Self {
+            rt,
+            system_state: Arc::new(Mutex::new(None)),
+            to_daemon_sender: Some(to_daemon_sender),
```

Comment:
Yeah, because in other place these values are taken, so they are no longer in this struct. And there is no other way to move only some fields of struct, which is what I wanted to be able to do.